syne_tune/optimizer/schedulers/multiobjective/multi_surrogate_multi_objective_searcher.py

```diff
@@ -0,0 +1,155 @@
+        num_initial_random_choices: int = DEFAULT_NUM_INITIAL_RANDOM_EVALUATIONS,
+        allow_duplicates: bool = False,
+        restrict_configurations: Optional[List[Dict[str, Any]]] = None,
+        clone_from_state: bool = False,
+        **kwargs,
+    ):
+        if mode is None:
+            mode = ["min" for item in metric]
+
+        super().__init__(
+            config_space,
+            metric=metric,
+            mode=mode,
+            points_to_evaluate=points_to_evaluate,
+            random_seed_generator=kwargs.get("random_seed_generator"),
+            random_seed=kwargs.get("random_seed"),
+        )
+        self.estimator = estimators
+        if scoring_class_and_args is None:
+            scoring_class_and_args = MultiObjectiveLCBRandomLinearScalarization
+
+        if not clone_from_state:
+            hp_ranges = make_hyperparameter_ranges(self.config_space)
+            self._create_internal(
+                hp_ranges=hp_ranges,
+                estimator=self.estimator,
+                acquisition_class=scoring_class_and_args,
+                num_initial_candidates=num_initial_candidates,
+                num_initial_random_choices=num_initial_random_choices,
+                initial_scoring="acq_func",
+                skip_local_optimization={name: True for name in self.estimator.keys()},
+                allow_duplicates=allow_duplicates,
+                restrict_configurations=restrict_configurations,
+                **kwargs,
+            )
+        else:
+            # Internal constructor, bypassing the factory
+            # Note: Members which are part of the mutable state, will be
+            # overwritten in ``_restore_from_state``
+            self._create_internal(**kwargs.copy())
+
+    def _update(self, trial_id: str, config: Dict[str, Any], result: Dict[str, Any]):
+        # Gather metrics
+        metrics = {name: result[name] for name in self._metric}
+
+        # Cost value only dealt with here if ``resource_attr`` not given
+        attr = self._cost_attr
+        cost_val = None
+        if attr is not None and attr in result:
+            cost_val = float(result[attr])
+            if self._resource_attr is None:
+                metrics[INTERNAL_COST_NAME] = cost_val
+
+        self.state_transformer.label_trial(
+            TrialEvaluations(trial_id=trial_id, metrics=metrics), config=config
+        )
+        if self.debug_log is not None:
+            _trial_id = self._trial_id_string(trial_id, result)
+            msg = f"Update for trial_id {_trial_id}: metric = {metrics:.3f}"
+            if cost_val is not None:
+                msg += f", cost_val = {cost_val:.2f}"
+            logger.info(msg)
+
+    def clone_from_state(self, state):
+        # Create clone with mutable state taken from 'state'
+        init_state = decode_state(state["state"], self._hp_ranges_in_state())
+        estimator = self.state_transformer.estimator
```

Comment:
See, you don't use `self.estimator` here. I'd prefer not to have `self.estimator` at all.

Reply:
Agreed, removed.